From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: talos-builder <talos-builder@users.noreply.github.com>
Date: Tue, 27 May 2026 22:00:00 +0000
Subject: [PATCH] [PATCH] Patched for Raspberry Pi 5

---
 Pkgfile                 | 8 ++++----
 kernel/prepare/pkg.yaml | 6 +++---
 2 files changed, 7 insertions(+), 7 deletions(-)

diff --git a/Pkgfile b/Pkgfile
index a272582..0000000 100644
--- a/Pkgfile
+++ b/Pkgfile
@@ -99,10 +99,10 @@ vars:
   kspp_sha256: 640b25473f86557d082f680b0f57653a69f4146e97825673bf582f9a4c8b7739
   kspp_sha512: 82dae1debbe94a3f82766a8cdbfe59ff8698d433175803458499b76c50b47b89e9280b677a9f16b2a44711badd6f4001aba951a41722a74dd51c03a18b8b9219
 
-  # renovate: datasource=git-tags extractVersion=^v(?<version>.*)$ depName=git://git.kernel.org/pub/scm/linux/kernel/git/stable/linux.git
-  linux_version: 6.18.24
-  linux_sha256: c207c557ce58103b4dda30e26da5203f3d8467c6dadc53d709f6d83ae1d1255f
-  linux_sha512: ad734fae886272f0c22853c885f618711baa2128b6447d6b996c884a16781893b73c65f5cf58087637a44d33ced1e5ecff9ead2fbca0bbeb7686dbbd4628de98
+  # renovate: datasource=git-tags extractVersion=^v(?<version>.*)$ depName=https://github.com/raspberrypi/linux.git
+  linux_version: stable_20250916
+  linux_sha256: 4639fe79697970b71cf708234b71d855a58ce5e28c377a4f7cf6ecbbe40647f3
+  linux_sha512: 5818d7ac495dc31a27c9b967bf146d291ab6fd06b99e819d9c685d5413472593cd99a4d7ce10218c45d7fb631adceed8e20806b1f2f9909277ce0de7ad204565
 
   # renovate: datasource=git-tags extractVersion=^libaio-(?<version>.*)$ depName=https://pagure.io/libaio.git
   libaio_version: 0.3.113
diff --git a/kernel/prepare/pkg.yaml b/kernel/prepare/pkg.yaml
index 6c6e61d..0000000 100644
--- a/kernel/prepare/pkg.yaml
+++ b/kernel/prepare/pkg.yaml
@@ -6,8 +6,8 @@ dependencies:
   - image: "{{ .LLVM_IMAGE }}:{{ .TOOLS_REV }}"
 steps:
   - sources:
-      - url: https://cdn.kernel.org/pub/linux/kernel/v{{ regexReplaceAll "(\\d+)(.\\d+)(\\.\\d+)?$" .linux_version "${1}" }}.x/linux-{{ .linux_version }}.tar.xz
-        destination: linux.tar.xz
+      - url: "https://github.com/raspberrypi/linux/archive/refs/tags/{{ .linux_version }}.tar.gz"
+        destination: linux.tar.gz
         sha256: "{{ .linux_sha256 }}"
         sha512: "{{ .linux_sha512 }}"
       - url: https://github.com/a13xp0p0v/kernel-hardening-checker/archive/{{ .kspp_ref }}.tar.gz
@@ -19,7 +19,7 @@ steps:
     prepare:
       - |
         mkdir -p /src
-        tar -xJf linux.tar.xz --strip-components=1 -C /src
+        tar -xzf linux.tar.gz --strip-components=1 -C /src
 
         cd /src
         make mrproper
-- 
2.43.0

